python_modules/dagster/dagster/_core/definitions/repository_definition/repository_definition.py

```diff
@@ -265,55 +264,18 @@ def asset_checks_defs_by_key(self) -> Mapping[AssetKey, "AssetChecksDefinition"]
         return self._repository_data.get_asset_checks_defs_by_key()
 
     def has_implicit_global_asset_job_def(self) -> bool:
-        """Returns true is there is a single implicit asset job for all asset keys in a repository."""
-        return self.has_job(ASSET_BASE_JOB_PREFIX)
+        return self.has_job(IMPLICIT_ASSET_JOB_NAME)
 
     def get_implicit_global_asset_job_def(self) -> JobDefinition:
-        """A useful conveninence method for repositories where there are a set of assets with
-        the same partitioning schema and one wants to access their corresponding implicit job
-        easily.
-        """
-        if not self.has_job(ASSET_BASE_JOB_PREFIX):
-            raise DagsterInvariantViolationError(
-                "There is no single global asset job, likely due to assets using "
-                "different partitioning schemes via their partitions_def parameter. You must "
-                "use get_implicit_job_def_for_assets in order to access the correct implicit job."
-            )
-
-        return self.get_job(ASSET_BASE_JOB_PREFIX)
+        return self.get_job(IMPLICIT_ASSET_JOB_NAME)
 
     def get_implicit_asset_job_names(self) -> Sequence[str]:
-        return [
-            job_name for job_name in self.job_names if job_name.startswith(ASSET_BASE_JOB_PREFIX)
-        ]
+        return [IMPLICIT_ASSET_JOB_NAME]
 
     def get_implicit_job_def_for_assets(
```

Comment:
`get_implicit_job_def_for_assets ` and `get_implicit_asset_job_names` will be removed in a downstream PR, which updates all its callers to use `get_implicit_global_asset_job_def`.